dispatch/BESCatalogList.h

```diff
@@ -98,13 +98,17 @@ class BESCatalogList: public BESObj {
 
     friend class BESCatalogListTest;
 
+    BESCatalogList();
+
 public:
     typedef std::map<std::string, BESCatalog *>::iterator catalog_iter;
```

Comment:
This can be rewritten with the C++14 _using_ syntax lie _using catalog_iter = td::map<std::string, BESCatalog *>::iterator;_

This is a general rule - we can replace the old C (c. 1970) with a _using_ declaration. It's not required, but I've been doing this as I come across typedefs.